### DISCUSSION TITLE
Buildpacks failing to create docker image of Spring Boot application when Docker Desktop version 4.31.x is installed

### DISCUSSION THREAD

**Discussion by Abomicode** [buildpacks/community#258] [Q&A] [answered]
Hello,
I am fairly new to Buildpacks so please do not mind if I mention something wrong. 
Issue: When creating a docker image of a Spring Boot application with the command (mvn spring-boot:build-image) having Docker Desktop version 4.31.x, I faced a consistent error -> Illegal char <:> at index 5: npipe:////./pipe/dockerDesktopLinuxEngine
When downgrading to Docker Desktop version 4.30.0 everything works fine and the docker images are also getting created successfully.
 Is anybody facing the same issue or if I am doing anything wrong? Thanks.

**Comment by natalieparellano** [comment]
Hi @Abomicode - thanks for raising this discussion. You may want to open an issue on https://github.com/spring-projects/spring-boot as based on your command, this appears to be an issue with the plugin. I'd be curious to know though how this turns out as the pack implementation may also need to be updated.

**Reply by Abomicode** [reply]
Hello @natalieparellano - thanks for your quick reply. As suggested, I will open an issue on the link provided above.

**Reply by Abomicode** [reply, marked as answer]
This is to help anyone facing the same issue regarding Docker image creation with command - mvn spring-boot:build-image with Docker Desktop Version - 4.31.x installed. Hope this helps.

Command: 
`docker context inspect`

Docker Desktop 4.30.0:
`[
  {
    "Name": "default",
    "Metadata": {},
    "Endpoints": {
      "docker": {
        "Host": "npipe:////./pipe/docker_engine",
        "SkipTLSVerify": false
      }
    },
    "TLSMaterial": {},
    "Storage": {
      "MetadataPath": "<IN MEMORY>",
      "TLSPath": "<IN MEMORY>"
    }
  }
]`

Docker Desktop 4.31.x:
`[
  {
    "Name": "desktop-linux",
    "Metadata": {
      "Description": "Docker Desktop"
    },
    "Endpoints": {
      "docker": {
        "Host": "npipe:////./pipe/dockerDesktopLinuxEngine",
        "SkipTLSVerify": false
      }
    },
    "TLSMaterial": {},
    "Storage": {
      "MetadataPath": "C:\\Users\\<username>\\.docker\\contexts\\meta\\fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e",
      "TLSPath": "C:\\Users\\<username>\\.docker\\contexts\\tls\\fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e"
    }
  }
]`

RCA: 
change from docker_engine to dockerDesktopLinuxEngine (by Docker Team) has caused this issue since the ResolvedDockerHost class (by Spring Boot Team) defaults to the former in absence of any other configuration.

Solution/Workaround (by Spring Boot Team):

Resolution for Maven (mention docker host explicitly in Maven pom file) - 
`<plugin>
	<groupId>org.springframework.boot</groupId>
	<artifactId>spring-boot-maven-plugin</artifactId>
	<configuration>
		<docker>
			<host>//./pipe/dockerDesktopLinuxEngine</host>
		</docker>
	</configuration>
</plugin>`

Resolution for Gradle (mention docker host explicitly in Gradle build file) -
`tasks.bootBuildImage {
    docker {
        host = "//./pipe/dockerDesktopLinuxEngine"
    }
}`
